### PR TITLE
Add Bus Mapping Support

### PR DESCRIPTION
**Pull request by sensokame**

### CODE REVIEW

**Comment by sensokame** [comment]
@kayoub5 can you do an initial review and see if I missed something? (the interface id value is always 2 am not sure if it's a mistake or that's the actual value from the blf)

File: src/app.cpp [hunk 1]
```diff
@@ -347,8 +349,8 @@ uint64_t calculate_startdate(Vector::BLF::File* infile) {
 }
 
 int main(int argc, char* argv[]) {
-	if (argc != 3) {
-		fprintf(stderr, "Usage %s [infile] [outfile]\n", argv[0]);
+	if (argc != 4) {
+		fprintf(stderr, "Usage %s [infile] [outfile] [mappingfile]\n", argv[0]);
```
Comment (kayoub5) [inline src/app.cpp:353]: Use same command line syntax as TECMP converter (you can copy the command line parsing code)

File: tests/mapping.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          "pkt_dir": "Tx"
+        }
+      },
+      {
+        "when": {
+          "chl_id": 3
+        },
+        "change": {
+          "inf_name": "can_three",
+          "pkt_dir": "Rx"
```
Comment (kayoub5) [inline tests/mapping.json:19]: You may want to update this file to match what we have in the BLF sample files

File: CMakeLists.txt [hunk 3]
```diff
@@ -55,6 +57,7 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
             COMMAND blf_converter 
                 "${CMAKE_CURRENT_LIST_DIR}/vector_blf/vector_blf/src/Vector/BLF/tests/unittests/events_from_${blf_test}.blf"
                 "${CMAKE_CURRENT_LIST_DIR}/tests/results/events_from_${blf_test}.pcapng"
+                "${CMAKE_CURRENT_LIST_DIR}/tests/mapping.json"
```
[the remark this answers is not in the document]
Comment (sensokame) [reply CMakeLists.txt:60]: the problem is that the command now NEEDS a mapping file

**Comment by sensokame** [comment]
@kayoub5 can you check the pull request now?

**Comment by kayoub5** [comment]
LGTM, is this PR still WIP ?

**Comment by sensokame** [comment]
i removed the wip status

**Pull request merged**